libs/index/odc/index/stac.py

```diff
@@ -100,7 +162,7 @@ def stac_transform(input_stac):
 
     stac_odc = {
         '$schema': 'https://schemas.opendatacube.org/dataset',
-        'id': deterministic_uuid,
+        'id': product_id,
```

Comment:
If my interpretation is wrong and `product_id` is not a UUID of a Dataset then it should not be here.

Reply:
Replaced variable name `product_id` with `deterministic_uuid `